services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/ExperimentalWaypointMetadata.java

```diff
@@ -0,0 +1,190 @@
+package com.mapbox.api.directions.v5.models;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.auto.value.AutoValue;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.TypeAdapter;
+import com.google.gson.annotations.SerializedName;
+import com.mapbox.api.directions.v5.DirectionsAdapterFactory;
+
+/**
+ * Object representing experimental value.
+ * <p>
+ * All available experimental values are subject to change at any time.
+ */
+@AutoValue
+public abstract class ExperimentalWaypointMetadata extends DirectionsJsonObject {
+
+  /**
+   * Create a new instance of this class by using the {@link ExperimentalWaypointMetadata.Builder}
+   * class.
+   *
+   * @return {@link ExperimentalWaypointMetadata.Builder} for creating a new instance
+   */
+  public static Builder builder() {
+    return new AutoValue_ExperimentalWaypointMetadata.Builder();
+  }
+
+  /**
+   * Object representing experimental value.
+   * <p>
+   * All available experimental values are subject to change at any time.
+   */
+  @Nullable
+  @SerializedName("type")
+  public abstract String type();
+
+  /**
+   * Object representing experimental value.
+   * <p>
+   * All available experimental values are subject to change at any time.
+   */
+  @Nullable
+  @SerializedName("name")
+  public abstract String name();
+
+  /**
+   * Object representing experimental value.
+   * <p>
+   * All available experimental values are subject to change at any time.
+   */
+  @Nullable
+  @SerializedName("charge_time")
+  public abstract Integer chargeTime();
+
+  /**
+   * Object representing experimental value.
+   * <p>
+   * All available experimental values are subject to change at any time.
+   */
+  @Nullable
+  @SerializedName("charge_to")
+  public abstract Integer chargeTo();
+
+  /**
+   * Object representing experimental value.
+   * <p>
+   * All available experimental values are subject to change at any time.
+   */
+  @Nullable
+  @SerializedName("plug_type")
+  public abstract String plugType();
+
+  /**
+   * Object representing experimental value.
+   * <p>
+   * All available experimental values are subject to change at any time.
+   */
+  @Nullable
+  @SerializedName("power_kw")
+  public abstract Integer powerKiloWatt();
```

Comment:
Thanks for bringing this up :+1: 

This response data will only be returned if the `electric engine` experimental parameter is used. Since it shouldn't be used in any type of production environment, we shouldn't need to worry about compatibility for our case. 

That said, I don't want to setup a dangerous precedent where an "experimental" property would be introduced and released in a stable version of the library and then potentially conflict with a server side response much, much later down the line, causing parsing exceptions.

I will take another pass at this, to make this safer. My current idea is to introduce a new annotation we discussed above but not only for documentation reasons but also to ensure that a stable version of `mapbox-java` cannot be released with an experimental property. Planning to use the annotation processing and `BuildConfig` to see if I can throw a build-time (preferably) or runtime exception in these cases. If this won't work, then maybe some alternative standalone script.